### PR TITLE
[OCPCLOUD-815] Use MachineSet controller from MAO image

### PR DESCRIPTION
**Pull request by JoelSpeed**
This changes the image for the MachineSet controller deployed in the `machine-api-controllers` deployment to use the `machine-api-operator` image instead of the `provider` image.

Also adds tests to check the output of `maoConfigFromInfrastructure` as this was not previously tested

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -9,6 +9,7 @@ require (
 	github.com/google/uuid v1.1.1
 	github.com/hashicorp/golang-lru v0.5.3 // indirect
 	github.com/imdario/mergo v0.3.8 // indirect
+	github.com/onsi/gomega v1.8.1
```
Comment (enxebre) [inline go.mod:12]: :face_with_raised_eyebrow:

**Comment by enxebre** [comment]
shall we include a commit to rename the binary in the kube deployment, the Dockerfile and the Make file? e.g manager -> machineSet

**Comment by JoelSpeed** [comment]
@enxebre Renaming seems good, have added another commit to rename to `machineset-controller`, PTAL

**Pull request merged**